crmsh/report/core.py

```diff
@@ -2,427 +2,489 @@
+            p = multiprocessing.Process(target=start_collector, args=(node, context))
             p.start()
             process_list.append(p)
     for p in process_list:
-            p.join()
-
-def dump_env():
-    """
-    this is how we pass environment to other hosts
-    """
-    env_dict = {}
-    env_dict["DEST"] = constants.DEST
-    env_dict["FROM_TIME"] = constants.FROM_TIME
-    env_dict["TO_TIME"] = constants.TO_TIME
-    env_dict["USER_NODES"] = constants.USER_NODES
-    env_dict["NODES"] = constants.NODES
-    env_dict["HA_LOG"] = constants.HA_LOG
-    # env_dict["UNIQUE_MSG"] = constants.UNIQUE_MSG
-    env_dict["SANITIZE_RULE_DICT"] = constants.SANITIZE_RULE_DICT
-    env_dict["DO_SANITIZE"] = constants.DO_SANITIZE
-    env_dict["SKIP_LVL"] = constants.SKIP_LVL
-    env_dict["EXTRA_LOGS"] = constants.EXTRA_LOGS
-    env_dict["PCMK_LOG"] = constants.PCMK_LOG
-    env_dict["VERBOSITY"] = int(config.report.verbosity) or (1 if config.core.debug else 0)
-
-    res_str = ""
-    for k, v in env_dict.items():
-        res_str += " {}={}".format(k, v)
-    return res_str
-
-def get_log():
-    """
-    get the right part of the log
-    """
-    outf = os.path.join(constants.WORKDIR, constants.HALOG_F)
-
-    # collect journal from systemd unless -M was passed
-    if constants.EXTRA_LOGS:
-        utillib.collect_journal(constants.FROM_TIME,
-                                constants.TO_TIME,
-                                os.path.join(constants.WORKDIR, constants.JOURNAL_F))
-
-    if constants.HA_LOG and not os.path.isfile(constants.HA_LOG):
-        if not is_collector(): # warning if not on slave
-            logger.warning("%s not found; we will try to find log ourselves", constants.HA_LOG)
-            constants.HA_LOG = ""
-    if not constants.HA_LOG:
-        constants.HA_LOG = utillib.find_log()
-    if (not constants.HA_LOG) or (not os.path.isfile(constants.HA_LOG)):
-        if constants.CTS:
-            pass  # TODO
-        else:
-            logger.warning("not log at %s", constants.WE)
-        return
+        p.join()
+
+
+def start_collector(node: str, context: Context) -> None:
+    """
+    Start collector at specific node
+    """
+    cmd = f"{constants.BIN_COLLECTOR} '{context}'"
+    err = ""
 
-    if constants.CTS:
-        pass  # TODO
+    if node == context.me:
+        code, out, err = ShellUtils().get_stdout_stderr(cmd)
     else:
-        try:
-            getstampproc = utillib.find_getstampproc(constants.HA_LOG)
-        except PermissionError:
-            return
-        if getstampproc:
-            constants.GET_STAMP_FUNC = getstampproc
-            if utillib.dump_logset(constants.HA_LOG, constants.FROM_TIME, constants.TO_TIME, outf):
-                utillib.log_size(constants.HA_LOG, outf+'.info')
+        node = f"{context.ssh_user}@{node}" if context.ssh_user else node
+        cmd = cmd.replace('"', '\\"')
+        cmd = f'ssh {constants.SSH_OPTS} {node} "{context.sudo} {cmd}"'
```

Comment:
Why is it needed to enclose `[sudo] <cmd>` with `"` here?

Reply:
Without `"`, got `Expecting property name enclosed in double quotes`

Reply:
Oh, `ssh` often messes up with those arguments.